src/server/cmd/pachctl/shell/shell.go

```diff
@@ -67,6 +95,13 @@ func (s *shell) executor(in string) {
 	cmd.Run()
 }
 
+// variables for caching completion calls
+var (
+	completionID string
+	suggests     []prompt.Suggest
+	cacheF       CacheFunc
+)
```

Comment:
Should these really be global?  This is going to make it harder to do tests - could they be moved into the `shell` instance?

Reply:
Good call, moved them into the shell instance.